src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts

```diff
@@ -226,9 +226,44 @@ export class TextSubmissionViewerComponent implements OnChanges {
             return escape(fileContent);
         }
 
+        if (this.exercise?.type === ExerciseType.PROGRAMMING) {
+            return this.buildFileContentWithHighlightedMatchesAsWholeLines(fileContent, matches);
+        }
         return this.buildFileContentWithHighlightedMatches(fileContent, matches);
     }
 
+    /**
+     * Builds the required HTML to highlight the lines with matches in the file content.
```

Comment:
Please extend the comment to describe _why_ only full lines are highlighted (in case of programming exercises).